lib/utils.js

```diff
@@ -178,31 +178,43 @@ function endOfMonthTimestamp() {
   return Date.UTC(curDate.getUTCFullYear(), curDate.getUTCMonth() + 1, 1, 0, 0, 0, 0);
 }
 
-function resolveElevatedParams(erlParams, bucketKeyConfig) {
+function resolveElevatedParams(erlParams, bucketKeyConfig, key) {
   // provide default values for elevated_limits unless the bucketKeyConfig has them
-  return {
+  const elevatedLimits = {
     ms_per_interval: bucketKeyConfig.ms_per_interval,
     size: bucketKeyConfig.size,
     erl_activation_period_seconds: 0,
     erl_quota: 0,
     erl_quota_interval: 'quota_per_calendar_month',
-    erl_is_active_key: 'defaultActiveKey',
-    erl_quota_key: 'defaultQuotaKey',
+    erl_is_active_key: 'ERLActiveKey',
+    erl_quota_key: 'ERLQuotaKey',
     ...erlParams,
     ...bucketKeyConfig.elevated_limits,
     erl_configured_for_bucket: !!(erlParams && bucketKeyConfig.elevated_limits?.erl_configured_for_bucket),
   };
+
+  elevatedLimits.erl_is_active_key = replicateHashtag(key, elevatedLimits.erl_is_active_key)
+  elevatedLimits.erl_quota_key = replicateHashtag(key, elevatedLimits.erl_quota_key)
+
+  return elevatedLimits;
 }
 
-class LimitdRedisConfigurationError extends Error {
-  constructor(msg, extra) {
-    super();
-    this.name = this.constructor.name;
-    this.message = msg;
-    Error.captureStackTrace(this, this.constructor);
-    if (extra) {
-      this.extra = extra;
-    }
+function replicateHashtag(baseKey, key) {
+  const idxOpenBrace = baseKey.indexOf('{')
+  if (idxOpenBrace < 0) {
+    return key + `:{${baseKey}}`;
+  }
+
+  const idxCloseBrace = baseKey.indexOf('}', idxOpenBrace)
+  if ( idxOpenBrace < 0 || idxCloseBrace <= idxOpenBrace ) {
```

Comment:
I think there is a typo there, you mean `idxCloseBrace` in the first check:

```
  if ( idxCloseBrace < 0 || idxCloseBrace <= idxOpenBrace ) {
```

But that is redundant because if we are here it means that `idxOpenBrace >= 0`, so the second condition `idxCloseBrace <= idxOpenBrace` will already catch that condition.

```suggestion
  if ( idxCloseBrace <= idxOpenBrace ) {
```

If I'm wrong and you wanted to write `if ( idxOpenBrace < 0 || ...)`, that check is also redundant, because if `idxOpenBrace < 0` we would have entered the previous check in line 204 and would have returned from this func.

Reply:
Oh, thanks. That was a last-second change as I was checking both open and close braces together, but after talking with @LeweyM we decided to have an early return if openbrace is not there, so this line should be as you suggest. Thx!